stream.py

```diff
@@ -87,500 +85,507 @@
 
 from operator import itemgetter, attrgetter
 
-zip = itertools.izip
+_filter = filter
+_map = map
+_reduce = functools.reduce
+
+try:
+    zip = itertools.izip
+except AttributeError:
+    pass
 
 try:
-	import multiprocessing
-	import multiprocessing.queues
-	_nCPU = multiprocessing.cpu_count()
+    import multiprocessing
+    _nCPU = multiprocessing.cpu_count()
 except ImportError:
-	_nCPU = 1
+    _nCPU = 1
 
 try:
-	Iterable = collections.Iterable
+    Iterable = collections.Iterable
 except AttributeError:
-	Iterable = object
+    Iterable = object
 
 try:
-	next
+    next
 except NameError:
-	def next(iterator):
-		return iterator.next()
+    def next(iterator):
+        return iterator.next()
 
 try:
-	from operator import methodcaller
+    from operator import methodcaller
 except ImportError:
-	def methodcaller(methodname, *args, **kwargs):
-		return lambda o: getattr(o, methodname)(*args, **kwargs)
+    def methodcaller(methodname, *args, **kwargs):
+        return lambda o: getattr(o, methodname)(*args, **kwargs)
 
 
-__version__ = '0.8'
+import pkg_resources
```

Comment:
FYI this is no longer available by default in Python 3.12 but I think it's ok for now.